scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py

```diff
@@ -5,115 +5,94 @@
 #     & Institut Laue - Langevin
 # SPDX - License - Identifier: GPL - 3.0 +
 from __future__ import (absolute_import, division, print_function)
+from re import findall
+from six import iteritems
 
-import mantid.simpleapi as mantid
+class MaxEnt(object):
 
-import Muon.GUI.Common.ADSHandler.workspace_naming as wsName
-from Muon.GUI.Common.contexts.muon_group_pair_context import get_default_grouping
-from Muon.GUI.Common.utilities.run_string_utils import run_list_to_string
+    def __init__(self, run, ws_freq):
+        print(run)
+        self.run = run
+        self.ws_freq = ws_freq
+
+
+class FFT(object):
+    # fft has two runs, one for Re and one for Im
+
+    def __init__(self, ws_freq_name, Re_run, Re, Im_run, Im, phasequad):
+        self.Re_run = Re_run
+        self.Im_run = Im_run
+        self.ws_freq_name = ws_freq_name
+        self.Re = Re
+        self.Im = Im
+        self.phasequad = phasequad
+
+
+FREQUENCY_EXTENSIONS ={"MOD":"mod", "RE":"Re", "IM":"Im", "MAXENT":"MaxEnt", "FFT":"FFT All" }
 
 
 class FrequencyContext(object):
+
     """
     A simple class for identifing the current run
     and it can return the name, run and instrument.
     The current run is the same as the one in MonAnalysis
     """
 
-    def __init__(self, context):
-        self.context = context
-        self.options = "None"
-        self.options = [item.replace(" ", "") for item in self.options]
-        self.N_points = 1
-        self.instrument = "None"
+    def __init__(self):
+        self._maxEnt_freq = {}
+        self._FFT_freq = {}
+        self.plot_type = "None"
+
+    @property
+    def window_title(self):
+        return "Frequency Domain Analysis"
+
+    def add_maxEnt(self, run, ws_freq):
+        self._maxEnt_freq[ws_freq.name()] = MaxEnt(run, ws_freq)
+
+    @property
+    def maxEnt_freq(self):
+        return [maxEnt for maxEnt in list(self._maxEnt_freq.keys())]
```

Comment:
would just list(self._maxEnt_freq.keys()) this do the same here?

Reply:
yes